lib/developer_portal/app/controllers/developer_portal/admin/account/passwords_controller.rb

```diff
@@ -1,14 +1,16 @@
 # frozen_string_literal: true
 
 class DeveloperPortal::Admin::Account::PasswordsController < ::DeveloperPortal::BaseController
+  include ThreeScale::BotProtection::Controller
+
   liquify prefix: 'password'
 
   skip_before_action :login_required
   before_action :find_provider
   before_action :find_user, :only => [:show, :update]
 
   def create
-    return redirect_to_request_password('Spam protection failed.') unless spam_check(buyer)
+    return redirect_to_request_password('Bot protection failed.') unless bot_check({ flash: false })
```

Comment:
This is not to change a password, it's to generate a new password when the user can't login because it forgot the old password, so they will never be logged in at this point.